app/response_processor.py

```diff
@@ -70,37 +70,37 @@ def process(self, msg, tx_id=None):
 
         self.logger = self.logger.bind(tx_id=self.tx_id)
 
-        response_type = str(decrypted_json.get("type"))
-
         try:
             self.validate_survey(decrypted_json)
         except ClientError:
             # If the validation fails, the message is to be marked "invalid"
             # and then stored. We don't then want to stop processing at this point.
-
             decrypted_json['invalid'] = True
             self.logger.info("Invalid survey data, skipping receipting")
-        else:
-            if response_type.find("feedback") == -1:
-                self.logger.info("Receipting survey")
-                self.send_receipt(decrypted_json)
-            else:
-                self.logger.info("Feedback survey, skipping receipting")
 
         self.store_survey(decrypted_json)
 
-        if 'invalid' in decrypted_json:
-            self.logger.error(
-                "Invalid survey response received, skipping notification")
+        if self._requires_receipting(decrypted_json):
+            self.send_receipt(decrypted_json)
 
-        elif response_type.find("feedback") == -1:
-            self.send_notification(decrypted_json.get("survey_id"))
-
-        else:
-            self.logger.info("Feedback survey, skipping notification")
+        if self._requires_downstream_processing(decrypted_json):
+            self.logger.info("Sending to downstream")
+            self.send_notification()
 
         self.logger.unbind("user_id", "ru_ref", "tx_id")
 
+    def _requires_receipting(self, decrypted_json):
+        if 'invalid' in decrypted_json:
```

Comment:
Can this check be done before the calls to `send_receipt` and `send_notification` and then exit the process method?